src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/access/AccessControlServiceTest.java

```diff
@@ -114,4 +116,54 @@ void get_roles_given_user_id_should_return_empty_list() {
             .isEqualTo(expectedAccessResponse);
 
     }
+
+    @Test
+    void should_succeed_and_return_access_control_response() {
+
+        final RoleAssignment mockedRoleAssignments = mock(RoleAssignment.class);
+        final UserInfo mockedUserInfo = mock(UserInfo.class);
+        final String idamToken = "someToken";
+
+        when(idamService.getUserInfo(idamToken))
+            .thenReturn(mockedUserInfo);
+
+        when(roleAssignmentService.getRolesForUser(mockedUserInfo.getUid(), idamToken))
+            .thenReturn(Collections.singletonList(mockedRoleAssignments));
+
+        Optional<AccessControlResponse> result = accessControlService.getAccessControlResponse(idamToken);
+
+        assertTrue(result.isPresent());
+        assertEquals(mockedUserInfo, result.get().getUserInfo());
+        assertEquals(mockedRoleAssignments, result.get().getRoleAssignments().get(0));
+
+        verify(idamService, times(1)).getUserInfo(idamToken);
+        verifyNoMoreInteractions(idamService);
+
+        verify(roleAssignmentService, times(1)).getRolesForUser(mockedUserInfo.getUid(), idamToken);
+        verifyNoMoreInteractions(roleAssignmentService);
+    }
+
+    @Test
+    void should_return_optional_empty_when_no_role_assignment_found() {
+
+        final UserInfo mockedUserInfo = mock(UserInfo.class);
+        final String idamToken = "someToken";
+
+        when(idamService.getUserInfo(idamToken))
+            .thenReturn(mockedUserInfo);
+
+        when(roleAssignmentService.getRolesForUser(mockedUserInfo.getUid(), idamToken))
+            .thenReturn(Collections.emptyList());
+
+        Optional<AccessControlResponse> result = accessControlService.getAccessControlResponse(idamToken);
+
+        assertTrue(result.isEmpty());
+
+        verify(idamService, times(1)).getUserInfo(idamToken);
+        verifyNoMoreInteractions(idamService);
+
+        verify(roleAssignmentService, times(1)).getRolesForUser(mockedUserInfo.getUid(), idamToken);
+        verifyNoMoreInteractions(roleAssignmentService);
+    }
```

Comment:
Maybe you can add one test when the service receives a `NoRoleAssignmentsFoundException` exception?  The result should still be an optional.empty().

Reply:
`should_return_optional_empty_when_thrown_NoRoleAssignmentsFoundException` method was added